### ISSUE TITLE
devtool=eval interferes with window.onerror, on Chrome.

### ISSUE THREAD

**Issue by savvopoulos** [webpack/webpack#5681]
**Do you want to request a *feature* or report a *bug*?**
Report a Bug.

**What is the current behavior?**
Bundling JS with webpack, using the option devtool: 'eval', adds `//@ sourceURL` tags that are prepended with `webpack:///` ; this, in turn, seems to confuse Chrome, which only sends a "Script Error" to window.error, because it thinks that the error was triggered by a script from a different origin.

**If the current behavior is a bug, please provide the steps to reproduce.**
https://github.com/savvopoulos/webpack-on-error
Note that you need an HTTP in order to properly repro the issue; file:/// seems to come with its own rules; http server included, needs python3.
* Build:
`./build.sh`
* Run:
`./serve.py`

Go to:
http://localhost:8000/index.html  - error looks good
Now go to:
http://localhost:8000/dist/index.html  - Script Error. :-(
Then: remove devtool: eval from the `webpack.config.js` - and rerun `./build.sh`.
Then, once again, go to:
http://localhost:8000/dist/index.html  - error looks good again!

**What is the expected behavior?**
I was able to get proper error reporting, even with devtool=eval, by manually removing the webpack:/// bit (e.g. open with vim and run `:%s/webpack:\/\/\///g`).

**If this is a feature request, what is motivation or use case for changing the behavior?**
(not a feature req)

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
This only happens in Chrome. Safari + Firefox are fine.

**Comment by sokra** [comment]
> This only happens in Chrome. Safari + Firefox are fine.

> this, in turn, seems to confuse Chrome

Sounds like a Chrome issue.

**Comment by savvopoulos** [comment]
Thanks @sokra for your response. I think you are right - I filed an issue with Chromium:
https://bugs.chromium.org/p/chromium/issues/detail?id=765909

Feel free to close. Btw, what is the purpose of the webpack:/// bit?

**Comment by ericziko** [comment]
If you add the following to your webPack.config it will mitigate this bug in chrome

 plugins: [
    // We use a source map plugin here with this special configuration
    // because if we do not - the window.onerror function does not work properly in chrome
    // and it swallows the errors because normally source maps have begin with webpack:///
    // here we are changing how the module file names are created
    // See this bug
    // https://bugs.chromium.org/p/chromium/issues/detail?id=765909
    new webpack.SourceMapDevToolPlugin({
      module: true,
      columns: false,
      moduleFilenameTemplate: info => { return `${info.resourcePath}?${info.loaders}` }
      // https://webpack.js.org/configuration/output/#output-devtoolmodulefilenametemplate
    })
]

**Comment by evilebottnawi** [comment]
Chrome bug (`Status: Assigned`). Workaround above (https://github.com/webpack/webpack/issues/5681#issuecomment-345861733) Thanks to everyone!

**Comment by adi518** [comment]
I'm having this exact issue, but the workaround doesn't work. I can't debug my JS logger because of this.

**Comment by Moghul** [comment]
Can we reopen this @evilebottnawi ? This is still an issue and is definitely not resolved.

Errors triggered with js in the .html page are logged just fine. Errors triggered in the bundles are logged with just "Script error", no file, no error, nothing.

I've tried the workaround above, and I've tried removing the plugin and using `devtool: 'source-map'`, and I've tried them together for the hell of it.

No attempt has worked. I use webpack cli to build my bundles, with or without --watch.

I am using webpack 4.17.1.
